### PR TITLE
8219 Show petitioner counsel modal to external and non docket users

### PR DESCRIPTION
**Pull request by mrinsin**

### CODE REVIEW

File: shared/src/business/useCases/updatePetitionerInformationInteractor.js [hunk 1]
```diff
@@ -237,14 +242,27 @@ exports.updatePetitionerInformationInteractor = async (
...
+      !practitioners.find(practitioner => practitioner.userId === user.userId)
+    ) {
+      throw new UnauthorizedError('Unauthorized for editing petition details');
+    }
+  } else {
+    if (!isAuthorized(user, ROLE_PERMISSIONS.EDIT_PETITIONER_INFO)) {
+      throw new UnauthorizedError('Unauthorized for editing petition details');
+    }
+  }
+
```
Comment (kkoskelin) [inline shared/src/business/useCases/updatePetitionerInformationInteractor.js]: Could we rearrange this logic a bit? maybe
```
const isRepresentingCounsel = practitioners.find(...);
const hasAuthorization = isRepresentingCounsel || isAuthorized(...);
if(!hasAuthorization) { throw... }
```

Comment (kkoskelin) [reply shared/src/business/useCases/updatePetitionerInformationInteractor.js]: Also, some tests to make sure this is doing what we want.

File: web-client/src/presenter/computeds/partiesInformationHelper.js [hunk 2]
```diff
@@ -42,16 +44,38 @@ export const partiesInformationHelper = (get, applicationContext) => {
...
+
+    const canEditRespondent = applicationContext
+      .getUtilities()
+      .isInternalUser(user.role);
+
+    const canEditParticipant = applicationContext
+      .getUtilities()
+      .isInternalUser(user.role);
 
     return {
```
Comment (kkoskelin) [inline web-client/src/presenter/computeds/partiesInformationHelper.js:68]: tests for these new return values?

File: web-client/src/presenter/computeds/viewCounselHelper.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+export const viewCounselHelper = get => {
+  const caseDetail = get(state.caseDetail);
+  const privatePractitioner = get(state.modal.contact);
+
+  const representingNames = privatePractitioner.representing.map(
+    representingId =>
+      caseDetail.petitioners.find(p => p.contactId === representingId).name,
+  );
+
+  return {
```
Comment (kkoskelin) [inline web-client/src/presenter/computeds/viewCounselHelper.js:12]: Would like to see tests for this, also.

**Review by kkoskelin — APPROVED** [review]
Once some test coverage is provided, 👍

**Review by rosiehsmith — APPROVED** [review]

**Pull request merged**